core/src/main/java/c/org/rajawali3d/annotations/InternalUseOnly.java

```diff
@@ -0,0 +1,22 @@
+package c.org.rajawali3d.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ *  Flag a type or method for internal/engine use only, and specifically not supported for use by engine clients as
+ *  part of its external API
+ *
+ *  Flagging a method applies to all of its implementations/extensions/overrides; flagging a type implies flagging
```

Comment:
What does `flagging a type implies flagging all methods in that type` mean?

Reply:
It means tagging at the class or interface level means all the methods in that type are also internal only, so we don't have to tag every method in an internal-only class or interface.  At your suggestion, I haven't used this anywhere.  Instead/for now, I started using "Internal" in the actual name, which certainly makes the point more visible.